feat: result table header styles

IrakliJani opened the pull request:
Closes: #2911 

### Description:
update styles for results table

<img width="1010" alt="CleanShot 2022-08-22 at 21 02 14@2x" src="https://user-images.githubusercontent.com/962095/185978016-7b455080-de93-45a9-aead-e0fb2457ec0a.png">

https://www.loom.com/share/6b989f79f8094acb982c3ca3f4d791b1

rephus commented on packages/frontend/src/components/LightdashVisualization/index.tsx:
```diff
@@ -1,5 +1,5 @@
 import { ChartType } from '@lightdash/common';
-import React, { FC } from 'react';
+import { FC } from 'react';
```
Unecessary  file change

rephus commented on packages/frontend/src/hooks/tableVisualization/getPivotDataAndColumns.ts:
```diff
@@ -109,6 +109,7 @@ const getPivotDataAndColumns = ({
             item: itemsMap[pivotDimension],
         },
     });
+
```
unecessary file change

rephus commented:
E2E tests are failing 
![image](https://user-images.githubusercontent.com/1983672/186133289-f1a0c445-dbc0-463a-9ea5-148f47d0a208.png)
It sounds like it can't find the header now that you have changed the format, so you should update the e2e test to match the new component.

rephus reviewed (changes requested):
I can't update the column name labels , I think that's what  `  getHeader,`
   ` getDefaultColumnLabel,` are for
![Peek 2022-08-23 12-20](https://user-images.githubusercontent.com/1983672/186134705-cc849966-2890-4db5-9d46-fd32bbc751d8.gif)

IrakliJani commented:
@rephus thanks for the review:

- fixed overriding header names
- fixed existing e2e tests
- added e2e to test hiding/showing tables from the config
- added e2e to test overriding column names
- bonus: updated chart type dropdown switcher to use Menu instead of custom button wrapper (see screens)


<img width="265" alt="CleanShot 2022-08-23 at 16 50 28@2x" src="https://user-images.githubusercontent.com/962095/186162350-b0da821b-eab7-4813-8cca-a1b23ab755c6.png">

vs old

<img width="210" alt="CleanShot 2022-08-23 at 16 51 35@2x" src="https://user-images.githubusercontent.com/962095/186162535-545fa661-2d9a-4e2f-98f8-8497d0f2c197.png">

rephus commented:
Everything looks ok to me, 

but this E2E test is failing 
![image](https://user-images.githubusercontent.com/1983672/186182233-142dfa1a-c47c-4c45-9921-5b3ddb43fc12.png)

Let me know if you need help debugging, you can download the videos or screenshots of cypress on github from here https://github.com/lightdash/lightdash/actions/runs/2911819474

IrakliJani commented:
@rephus I thought it was a flaky test. then I thought I was going mad. I think what was happening is that you added two extra tables to the jaffle dashboard to test the PR. and e2e also uses jaffle dashboard.

rephus commented:
> @rephus I thought it was a flaky test. then I thought I was going mad. I think what was happening is that you added two extra tables to the jaffle dashboard to test the PR. and e2e also uses jaffle dashboard.

Damn, I'm so sorry, you were right, I did some manual testing on jaffle dashboard and I broke the tests without noticing >_<

My fault, sorry. 

Everything looks great then. Let's :sheep: it ! Great work.
